### ISSUE TITLE
Default prefix "normalized" not working

### ISSUE THREAD

**Issue by dheerajbhaskar** [slhck/ffmpeg-normalize#20]
## Issue 
The operation errors out if no prefix is given

## Expected output
When no prefix is given, the default "normalized" should be assumed

## Steps to reproduce
1. ffmpeg-normalize.exe --merge --prefix --verbose VID_20170119_110840.mp4

## Console output

    ffmpeg-normalize.exe --merge --prefix --verbose VID_20170119_110840.mp4
    2017-01-19 14:08:30,884 - ffmpeg_normalize - ERROR - error running command: "C:\Users\Dheeraj\include_in_path\ffmpeg" -y -i "VID_20170119_110840.mp4" -strict -2 -vcodec copy -af "volume=11.4dB" "--verbose-VID_20170119_110840.mp4"
    2017-01-19 14:08:30,884 - ffmpeg_normalize - ERROR - ffmpeg version N-83152-gf7e9275 Copyright (c) 2000-2017 the FFmpeg developers
      built with gcc 5.4.0 (GCC)
      configuration: --enable-gpl --enable-version3 --enable-cuda --enable-cuvid --enable-d3d11va --enable-dxva2 --enable-libmfx --enable-nvenc --enable-avisynth --enable-bzlib --enable-fontconfig --enable-frei0r --enable-gnutls --enable-iconv --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libfreetype --enable-libgme --enable-libgsm --enable-libilbc --enable-libmodplug --enable-libmp3lame --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenh264 --enable-libopenjpeg --enable-libopus --enable-librtmp --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libtheora --enable-libtwolame --enable-libvidstab --enable-libvo-amrwbenc --enable-libvorbis --enable-libvpx --enable-libwavpack --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxavs --enable-libxvid --enable-libzimg --enable-lzma --enable-decklink --enable-zlib
      libavutil      55. 43.100 / 55. 43.100
      libavcodec     57. 74.100 / 57. 74.100
      libavformat    57. 62.100 / 57. 62.100
      libavdevice    57.  2.100 / 57.  2.100
      libavfilter     6. 69.100 /  6. 69.100
      libswscale      4.  3.101 /  4.  3.101
      libswresample   2.  4.100 /  2.  4.100
      libpostproc    54.  2.100 / 54.  2.100
    Unrecognized option '-verbose-VID_20170119_110840.mp4'.
    Error splitting the argument list: Option not found

**Comment by slhck** [comment]
You can now set `--no-prefix` if you don't want a prefix.

Since the `--prefix` option requires a parameter, it can't be empty.

**Comment by dheerajbhaskar** [comment]
@slhck I actually wanted a prefix but wanted to use the default one i.e. `normalized`. I think we're not on the same page?

How is `--no-prefix` different from leaving out the option `--prefix` ?

Let me see if I can grok your commit

**Comment by dheerajbhaskar** [comment]
@slhck 
Oh wait! now I think I get it. Please tell me if my understanding is correct.

You use `normalized` as the prefix by default even without having the --prefix flag. I understood differently from the documentation in the README :)

**Comment by slhck** [comment]
Yes, the way those CLI options work, you can override the default prefix (which is `normalized`) with the `--prefix` option.

But since it wasn't possible to specify an empty prefix using such an option (`--prefix ""` would lead to the same error you wrote about in your initial report), I added a `--no-prefix` option.

I think back when I developed the tool, I wanted every file to have a prefix out of fear of overwriting the original file, well, now I guess I should add a check if `input_file == output_file` …

**Comment by dheerajbhaskar** [comment]
@slhck my recommendation, if I may, would be to just update the documentation. The fact you'd mentioned 'default' against a flag made me assume that this was the default for that flag. 

I foresee using your tool again, so I can provide this minor PR to the documentation 

As to the code changes - of having additional checks to see if input name is same as output - I think they are unnecessary or serve no purpose unless I've misunderstood you. 

So, let me know if if you'd like the PR and IMHO additional code changes are not needed :-)

**Comment by dheerajbhaskar** [comment]
If we were to rethink this, I'd have suggested:

By default doing what --no-prefix does and having a --prefix option to specify a default prefix and a --prefix <new-prefix> variant to accept custom prefix. 

Having said that I understand that having a default prefix is desirable (to prevent overwriting the source file by mistake). So just updating the documentation to make it clear that this is a global default (not related to a flag's default) and make no other code changes is what I would do. 

I just rephrased my previous comment to make my comment a little clearer. I've no new points here

**Comment by slhck** [comment]
Thanks for your suggestions. I absolutely understand your confusion with the documentation. However, in light of what is common and standard behavior for Unix tools – and what an argument parsing software can implement from a logical standpoint – I think it's is quite unambiguous and the behavior is well-defined.

Let me explain (slow Friday here at the office):

1. Remember that an option is always something *optional*. If you do not set an option, a default will be assumed. Of course, you must tell the user what that default is: it is shown in brackets in the documentation (i.e., `normalized`). Only when you want to change such a default, you set the option. This is how all well-behaving Unix tools work. (For example, see how [the `less` program](https://linux.die.net/man/1/less) tells you about what it does by default unless you set some option. Or, think about `git pull <url>` – it chooses unlimited depth for you, but you can also do `git pull --depth 0 <url>` to specify that you only want one commit.)

2. When you specify an option-argument in a utility documentation (e.g. `--prefix <prefix>`, with those brackets), the option-argument *must* be given by the user when he or she is specifying the option. There cannot be an option plus option-argument without its argument. You can check the [POSIX Utility Conventions](http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap12.html) for more info on that.

To summarize – and taking up your suggestion – there cannot be an option `--prefix` that specifies a default *and* an option `--prefix` that allows you to pass some other prefix. The argument parser must know if `--prefix` takes an option argument or not. It cannot be specified as both at the same time.

I would be hesitant to change the tool's behavior to remove the default `normalized` prefix, since it may break existing scripts (including some I have internally). The `--no-prefix` option should give you the possibility to tackle all use-cases, I hope.

**Comment by slhck** [comment]
By the way, this tool uses [`docopt`](https://github.com/docopt/docopt) and its syntax for specifying options, and I haven't deviated from it. If you find it ambiguous, it's actually a much broader issue having to do with Unix and the way `docopt` chose to implement that.
